snapcraft/_store.py

```diff
@@ -59,25 +54,42 @@ def logout():
 
 
 def upload(snap_filename):
-    if not os.path.exists(snap_filename):
-        raise FileNotFoundError(snap_filename)
-    else:
-        snap_name = _get_name_from_snap_file(snap_filename)
-        logger.info('Uploading existing {}.'.format(snap_filename))
+    logger.info('Uploading existing {}.'.format(snap_filename))
 
-        storeapi.upload(snap_filename, snap_name)
+    try:
+        store = storeapi.StoreClient()
+        result = store.upload(snap_filename)
+    except storeapi.InvalidCredentialsError:
+        logger.info('No valid credentials found.'
+                    ' Have you run "snapcraft login"?')
+        return
```

Comment:
you are right, good catch. Also it should be error, not info. I made the fix, and added a test for the raise.